packages/mpc-tls/prover/src/proof/redact.rs

```diff
@@ -0,0 +1,131 @@
+use hyper::{
+    header::HeaderValue,
+    http::{Request, Response},
+    HeaderMap,
+};
+use serde_json::{from_str, Value};
+
+#[derive(Debug)]
+pub struct HTTPParts {
+    pub headers: HeaderMap<HeaderValue>,
+    pub body: Option<String>,
+}
+
+impl HTTPParts {
+    pub fn get_header(&self, header_name: String) -> Option<String> {
+        println!("header-name: {}", header_name);
+        return self
+            .headers
+            .get(header_name)
+            .map(|header_value: &HeaderValue| header_value.to_str().unwrap_or("").to_string());
+    }
+
+    pub fn get_body(&self, body_path: String) -> Option<String> {
+        let json_str = self.body.clone().unwrap_or(r#"{}"#.to_string());
+        // Parse the JSON string
+        let json_value: Value = from_str(json_str.as_str()).ok()?;
+        let result = body_path.split('.').try_fold(&json_value, |current, part| {
+            if let Some(index) = part.parse::<usize>().ok() {
+                current.get(index)
+            } else {
+                current.get(part)
+            }
+        });
+
+        result
+            .cloned()
+            .map(|x| x.to_string().trim_matches('"').to_string())
+    }
+}
+
+impl From<&Request<String>> for HTTPParts {
+    fn from(request: &Request<String>) -> Self {
+        Self {
+            headers: request.headers().clone(),
+            body: Some(request.body().clone()).filter(|s| !s.is_empty()),
+        }
+    }
+}
+
+impl From<&Response<String>> for HTTPParts {
+    fn from(response: &Response<String>) -> Self {
+        Self {
+            headers: response.headers().clone(),
+            body: Some(response.body().clone()).filter(|s| !s.is_empty()),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct Redactor {
+    request: HTTPParts,
+    response: HTTPParts,
+}
+
+impl Redactor {
+    pub fn new(request: &Request<String>, response: &Response<String>) -> Self {
+        Self {
+            request: HTTPParts::from(request),
+            response: HTTPParts::from(response),
+        }
+    }
+
+    pub fn get_parameter(&self, parameter_path: String) -> Option<String> {
+        let parts: Vec<&str> = parameter_path.split(':').collect();
+
+        let response = match parts.as_slice() {
+            ["res", "header", header_name] => self.response.get_header(header_name.to_string()),
+            ["res", "body", property_path] => self.response.get_body(property_path.to_string()),
+            ["req", "header", header_name] => self.request.get_header(header_name.to_string()),
+            ["req", "body", property_path] => self.request.get_body(property_path.to_string()),
```

Comment:
There is no such thing as a `req:body` - right?

Reply:
not at the moment, but giving it some thought now and it would be a nice to have. i will include it.